File: group-anagrams-49/group_anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:

        if len(strs) == 0:
            return []
        _map = {}

        for _str in strs:
            letters = [0] * 26

            for c in _str:
                index = ord(c) - ord('a')
                letters[index]+=1
            key = tuple(letters)

            if key in _map:
                _map[key].append(_str)
            else:
                _map[key] = [_str]

        return list(_map.values())
            

        

File: group-anagrams-49/group_anagrams.rb
# @param {String[]} strs
# @return {String[][]}
def group_anagrams(strs)
    map = {}
    strs.each do |str|
        letters = [0]*26
        str.split("").each do |c|
            i = c.ord - 'a'.ord
            letters[i]+=1
        end

        key = letters.to_s
        if map.has_key? key
            map[key].append(str)
        else
            map[key] = [str]

        end
    end

    map.values
end